functorch/csrc/BatchRulesViews.cpp

```diff
@@ -272,6 +272,17 @@ std::tuple<Tensor, optional<int64_t>> select_batching_rule(const Tensor& self, o
   return std::make_tuple(result, 0);
 }
 
+std::tuple<Tensor, optional<int64_t>> _reshape_alias_batch_rule(const Tensor& self, optional<int64_t> bdim, const IntArrayRef shape, const IntArrayRef strides) {
+  (void) strides;
+  if (!bdim) {
+    return std::make_tuple(self.reshape(shape), nullopt);
+  }
+
+  auto new_shape = shape.vec();
```

Comment:
nit: new_shape is a std::vector and std::vector uses dynamic allocations. We should prefer to use SmallBuffer or SmallVector whenever possible.

Example usage: https://github.com/facebookresearch/functorch/blob/43889f5e8ec22f1b5de914232aa9284911adb1fb/functorch/csrc/BatchRulesViews.cpp#L282

Concretely, we want to preallocate a SmallBuffer<int64_t, 5> with (shape.size() + 1) and then std::copy into it and finally use that as the argument to reshape